wandb/sdk/internal/internal_api.py

```diff
@@ -2454,7 +2457,28 @@ def commit_artifact(self, artifact_id):
         }
         """
         )
-        response = self.gql(mutation, variable_values={"artifactID": artifact_id})
+
+        def check_retry(e: Any) -> bool:
+            if util.no_retry_auth(e):
+                return True
+            if hasattr(e, "exception"):
+                e = e.exception
+            if isinstance(e, requests.HTTPError) and e.response.status_code == 409:
+                logger.error("conflict when trying to commit artifact; retrying")
+                return True
+            return False
+
+        execute = retry.Retry(
```

Comment:
i don't think you need to do this... you can use the `self.gql` and just pass in your arguments, something like this:
```
response = self.gql(mutation, retry_timedelta=..., retryable_exceptions=..., check_retry_fn=..., variable_values=...)
```
if you look at the implementation of `retry.Retry`: we overwrite the default argument when we call it see this method here: https://github.com/wandb/client/blob/master/wandb/sdk/lib/retry.py#L70-L93

Reply:
...oh! You're totally right! Thanks, fixed!